### ISSUE TITLE
Hacmini Installation Security boot will display installation failure

### ISSUE THREAD

**Issue by hujun52** [osy/HaC-Mini#1016]
Because the automatic update of MACOS is required, so when installing HACMINI, select the safe boot option, and then show installation failure, try many times, will be like this, do not know where the problem is!

**Comment by Medizinmann-MD** [comment]
You need to provide a litte more information to get help here.

But I make a wild guess and say the process for the personalization of your drive failed.

You might find the solution here:
https://osy.gitbook.io/hac-mini-guide/post-installation/support#cannot-boot-when-secure-boot-is-enabled

**Comment by Simanias** [comment]
I've got exact the same problem with version 2.0. 
Used to have secure boot working but the installer fails and from there on, secure boot is messed up. After the restart the volume is messed up and it won't start. 
I had to reinstall without choosing the secure boot option to make the system bootable again. 

I've then tried fix it by going into recovery partition and create new keys, but the installer just keeps hanging. I have also reinstalled the MacOS system completely but still no joy.

**Comment by hujun52** [comment]
> 您需要提供一点更多信息才能在这里获得帮助。
> 
> 但我做了一个疯狂的猜测，并说你的驱动器个性化过程失败了。
> 
> 您可以在这里找到解决方案： https://osy.gitbook.io/hac-mini-guide/post-installation/support#cannot-boot-when-secure-boot-is-enabled

Hello, thank you for your reply. It was hacmini that prompted an unknown error at the last minute of installation, and the installation failed! If "enable boot security option" is not selected, the installation will be normal

So I don't know what the problem is!

**Comment by Medizinmann-MD** [comment]
Okay, so you got a problem  with the HaCMini.pkg itself? 

Do you mean for patching the OSX-Installer or when you install it on the finished MacOS-Install?

You actually need to activate secure boot later when you run the HaCMini.pkg on the finished install - and don't forget to install Xcode beforehand.

**Comment by hujun52** [comment]
Hello, I mean: hacmini was installed before PKG does not check the option of safe boot boot. Because MacOS needs automatic update, you need to reinstall hacmini.pkg to boot safely to get the automatic update function of MacOS. However, install hacmini After the PKG program checks safe boot, it will prompt that the installation failed, and an error will appear that the installation cannot be completed correctly!

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

________________________________
发件人: Medizinmann-MD ***@***.***>
发送时间: Monday, September 5, 2022 8:00:02 PM
收件人: osy/HaC-Mini ***@***.***>
抄送: hujun52 ***@***.***>; Author ***@***.***>
主题: Re: [osy/HaC-Mini] Hacmini Installation Security boot will display installation failure (Issue #1016)


Okay, so you got a problem with the HaCMini.pkg itself?

Do you mean for patching the OSX-Installer or when you install it on the finished MacOS-Install?

You actually need to activate secure boot later when you run the HaCMini.pkg on the finished install - and don't forget to install Xcode beforehand.

―

**Comment by Medizinmann-MD** [comment]
Again - we need more info to help you!

And again- did you install Xcode beforehand?

**Comment by hujun52** [comment]
All right, thank you! I'll go back and try it. Xcode has not been installed before!发自我的iPhone------------------ Original ------------------From: Medizinmann-MD  ***@***.***>Date: Tue,Sep 6,2022 3:51 PMTo: osy/HaC-Mini ***@***.***>Cc: hujun52 ***@***.***>, Author ***@***.***>Subject: Re: [osy/HaC-Mini] Hacmini Installation Security boot will displayinstallation failure (Issue #1016)
Again - we need more info to help you!
And again- did you install Xcode beforehand?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/osy/HaC-Mini/issues/1016#issuecomment-1237785398",
"url": "https://github.com/osy/HaC-Mini/issues/1016#issuecomment-1237785398",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by caiguo37** [comment]
The security boot option is a bit confusing.
After 12.2, does security boot have to be turned on for installing OSX, or can you still turn it off as long as you don't do OTA?

**Comment by osy** [comment]
Yes you need it on for OTA. You can keep it off if you don’t need OTA.

**Comment by caiguo37** [comment]
Thank you, that makes sense now.
